### PR TITLE
Admin changes to application pages

### PR DESCRIPTION
**Pull request by hammerfunk**

### CODE REVIEW

File: app/assets/javascripts/admin/comments.js.coffee [hunk 1]
```diff
@@ -1,10 +1,11 @@
 ready = ->
-  $('body').on 'click', '#build_new_comment', (e)->
-    e.preventDefault()
-    unless $('#new_comment').length
-      $.get $(this).attr('href'), (data)->
-        $('.comments-container').append(data)
-      , 'html'
+  buildComment = ->
+    unless $("#new_comment").length
+      $.get $(".comments-container").attr("data-comment-form"), (data)->
```
Comment (matthewford) [inline app/assets/javascripts/admin/comments.js.coffee]: Function arrow (->) must be spaced properly

File: app/assets/stylesheets/admin/base.scss [hunk 2]
```diff
@@ -600,250 +587,85 @@
...
-  margin-right: 50px;
-
-  .sidebar-open & {
-    width: 70%;
-    padding-right: 15px;
-  }
+.comment {
+  position: relative;
+  padding: 15px 0;
+  border-bottom: 1px solid #999;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#999` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 3]
```diff
@@ -600,250 +587,85 @@
...
-    text-align: center;
-    text-decoration: none;
-
-    &:hover {
-      opacity: 0.9;
-    }
+.comment-action {
+  display: block;
+  margin-top: 10px;
+  color: #999;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#999` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 4]
```diff
@@ -600,250 +587,85 @@
...
-      width: 12px;
-      height: 15px;
-      margin-top: 0;
-      margin-left: 7px;
-      background-size: 12px 15px;
-    }
+  &,
+  &:hover,
+  &:focus {
+    color: #999;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#999` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 5]
```diff
@@ -600,250 +587,85 @@
...
-    .sidebar-open & {
-      background-image: image-url("icon-flag-sm.png");
-
-      @include is-retina {
-        background-image: image-url("icon-flag-sm@2.png");
-      }
+    &,
+    &:hover,
+    &:focus {
+      color: #333;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#333` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 6]
```diff
@@ -876,6 +698,7 @@
   left: 0;
   width: 100%;
   background: rgba(0, 0, 0, 0.8);
+  color: #fff;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#fff` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 7]
```diff
@@ -950,9 +775,151 @@
...
   }
 
-  .pull-right {
-    margin-top: 0.3em;
+  small {
+    display: inline-block;
+    width: 100%;
+    margin-top: 10px;
+    padding-left: 25px;
+    color: #555;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#555` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 8]
```diff
@@ -950,9 +775,151 @@
...
+    font-size: 14px;
     font-style: italic;
     text-transform: none;
   }
 }
+
+.panel-parent {
+  > .panel-heading {
+    background-color: #706f6f;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#706f6f` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 9]
```diff
@@ -950,9 +775,151 @@
...
+
+    > .panel-title a {
+      background-image: image-url("icon-arrow-collapse-down-white.png");
+      color: #fff;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#fff` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/admin/base.scss [hunk 10]
```diff
@@ -950,9 +775,151 @@
...
+
+    @include is-retina {
+      background-image: image-url("icon-grade-neutral@2.png");
+    }
+  }
+
+  .rag-neutral {
+    &,
+    a {
+      color: #333 !important;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#333` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>!important should not be used

File: app/assets/stylesheets/admin/base.scss [hunk 11]
```diff
@@ -950,9 +775,151 @@
...
+      @include is-retina {
+        background-image: image-url("icon-grade-neutral@2.png");
+      }
+    }
+  }
+
+  .rag-positive {
+    &,
+    a {
+      color: #93c355 !important;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#93c355` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>!important should not be used

File: app/assets/stylesheets/admin/base.scss [hunk 12]
```diff
@@ -950,9 +775,151 @@
...
+      @include is-retina {
+        background-image: image-url("icon-grade-greenb@2.png");
+      }
+    }
+  }
+
+  .rag-average {
+    &,
+    a {
+      color: #f39200 !important;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#f39200` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>!important should not be used

File: app/assets/stylesheets/admin/base.scss [hunk 13]
```diff
@@ -950,9 +775,151 @@
...
+      @include is-retina {
+        background-image: image-url("icon-grade-amber@2.png");
+      }
+    }
+  }
+
+  .rag-negative {
+    &,
+    a {
+      color: #e30613 !important;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Color literals like `#e30613` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>!important should not be used

File: app/assets/stylesheets/admin/base.scss [hunk 14]
```diff
@@ -600,250 +587,85 @@
   }
 
 // Comments sidebar
-.col-content,
-.col-sidebar {
-  .sidebar-open & {
-    float: left;
-  }
+.comments-container textarea {
+  display: block !important;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: !important should not be used

File: app/assets/stylesheets/admin/base.scss [hunk 15]
```diff
@@ -950,9 +775,151 @@
...
+// RAG dropdown
+.btn-rag {
+  .icon-rag {
+    position: absolute;
+    top: 50%;
+    left: 0;
+    width: 20px;
+    height: 20px;
+    margin-top: -10px;
+    margin-right: 0 !important;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: !important should not be used

File: app/assets/stylesheets/admin/base.scss [hunk 16]
```diff
@@ -950,9 +775,151 @@
...
+
+  .rag-neutral .icon-rag {
+    background-image: image-url("icon-grade-neutral.png");
+
+    @include is-retina {
+      background-image: image-url("icon-grade-neutral@2.png");
+    }
+  }
+
+  .rag-neutral {
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Merge rule `.rag-neutral` with rule on line 832

File: app/assets/stylesheets/admin/base.scss [hunk 17]
```diff
@@ -335,6 +290,37 @@
   text-transform: uppercase;
 }
 
+.icon-view {
+  display: inline-block;
+  width: 30px;
+  height: 30px;
+  text-align: left;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Properties should be ordered display, width, height, background-image, background-repeat, background-position, background-size, text-align, text-indent, opacity

File: app/decorators/form_answer_decorator.rb [hunk 18]
```diff
@@ -14,6 +14,11 @@ def award_type
     object.class::AWARD_TYPE_FULL_NAMES[object.award_type]
   end
 
+  def award_type_short_name
+    award_titles = { "Innovation" => "Innovation", "International Trade"=> "Intl Trade", "Sustainable Development"=> "Sus Dev", "Enterprise promotion" => "QAEP" }
```
Comment (matthewford) [inline app/decorators/form_answer_decorator.rb]: Line is too long. [162/100]<br>Surrounding space missing for operator '=>'.

File: app/assets/stylesheets/admin/_variables.scss [hunk 19]
```diff
@@ -0,0 +1,10 @@
+$black-true:      #000;
+$black:           #333;
+$grey-dark:       #555;
+$grey-panel-dark: #706f6f;
+$grey:            #999;
+$grey-panel:      #dadad9;
+$white-true:      #fff;
+$green:           #93c355;
+$amber:           #f39200;
+$red:             #e30613;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/_variables.scss]: Files should end with a trailing newline

File: app/assets/stylesheets/admin/header-footer.scss [hunk 20]
```diff
@@ -13,7 +13,7 @@
 
     .navbar-brand,
     .navbar-nav > li > a {
-      color: #fff;
+      color: $white-true;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/header-footer.scss]: Properties should be ordered background, color, text-transform

File: app/assets/stylesheets/application-admin.scss [hunk 21]
```diff
@@ -12,6 +12,7 @@ $yellow: #ffbf47;
 $button-colour: #00823b;
 @import "frontend/password_strength";
 
+@import "admin/_variables";
```
Comment (matthewford) [inline app/assets/stylesheets/application-admin.scss]: Imported partial `admin/_variables` should be written as `admin/variables`

File: app/assets/stylesheets/admin/_variables.scss (hunk 19 above)
Comment (hammerfunk) [inline app/assets/stylesheets/admin/_variables.scss]: It does now :unamused:

File: app/decorators/form_answer_decorator.rb [hunk 22]
```diff
@@ -14,6 +14,14 @@ def award_type
     object.class::AWARD_TYPE_FULL_NAMES[object.award_type]
   end
 
+  def award_type_short_name
+    award_titles = { "Innovation" => "Innovation",
+                     "International Trade"=> "Intl Trade",
```
Comment (matthewford) [inline app/decorators/form_answer_decorator.rb]: Surrounding space missing for operator '=>'.

File: app/decorators/form_answer_decorator.rb [hunk 23]
```diff
@@ -14,6 +14,14 @@ def award_type
...
+                     "Sustainable Development"=> "Sus Dev",
```
Comment (matthewford) [inline app/decorators/form_answer_decorator.rb]: Surrounding space missing for operator '=>'.

File: app/assets/stylesheets/admin/base.scss (hunk 5 above)
Comment (hammerfunk) [inline app/assets/stylesheets/admin/base.scss]: This is also fixed...

File: app/decorators/form_answer_decorator.rb [hunk 24]
```diff
@@ -14,6 +14,14 @@ def award_type
     object.class::AWARD_TYPE_FULL_NAMES[object.award_type]
   end
 
+  def award_type_short_name
+    award_titles = { "Innovation" => "Innovation",
+                     "International Trade" => "Intl Trade",
+                     "Sustainable Development" => "Sus Dev",
+                     "Enterprise promotion" => "QAEP" }
+    award_titles[object.class::AWARD_TYPE_FULL_NAMES[object.award_type]]
+  end
```
Comment (RuslanHamidullin) [inline app/decorators/form_answer_decorator.rb]: I suppose would be great to define these list as constant as it also used in view (app/views/admin/form_answers/index.html.slim)

So can be like:

```
class FormAnswerDecorator < ApplicationDecorator
  SELECT_BOX_LABELS = ...
  AWARD_TITLES =  {
    "Innovation" => "Innovation",
    "International Trade" => "Intl Trade",
    "Sustainable Development" => "Sus Dev",
    "Enterprise promotion" => "QAEP" 
  }

...bla bla

def award_type_short_name
    AWARD_TITLES[object.class::AWARD_TYPE_FULL_NAMES[object.award_type]]
end
```

and then use it on view in app/views/admin/form_answers/index.html.slim

File: app/views/admin/form_answers/index.html.slim [hunk 25]
```diff
@@ -44,7 +44,11 @@
                           label data-value=v
                             input type="checkbox"
                             span.label-contents
-                              = key
+                              - award_titles = { "Innovation" => "Innovation",
+                                                 "International Trade"=> "Intl Trade",
+                                                 "Sustainable Development"=> "Sus Dev",
+                                                 "QAEP" => "QAEP" }
+                              = award_titles[key]
```
Comment (RuslanHamidullin) [inline app/views/admin/form_answers/index.html.slim]: and then here:
```
= FormAnswerDecorator::AWARD_TITLES[key]
```

**Comment by RuslanHamidullin** [comment]
Looks good :+1: Just small not about award titles

**Comment by RuslanHamidullin** [comment]
Great :+1:

**Pull request merged**